DOC-56 Import Technology Providers Playbook (#110)
Ed-Fi-Alliance-OSS/ed-fi-alliance-oss.github.io: docs/getting-started/provider-playbook/technology-providers-implementation/ed-fi-api-fundamentals/enumerations-descriptors.md

  }
```

Note that a Resource ID ("id") is provided here, consistent with other Ed-Fi REST API resources.

## Ed-Fi, Provider, and Agency Values: What to Expect and What to Use

Interoperability is complex, and enumeration values are part of this complexity. While it seems like the goal should be to coordinate the whole K-12 ecosystem to employ use a single set of values for each core K-12 concept, the reality is that education agencies and technology providers must adapt values to local needs and conditions.

For agencies, such adaptation often reflects important local programs and priorities. Further, use of some sets of values might be mandated by (or reflect) state or local policy. These make adoption of a global set problematic.

For their part, technology providers often have highly refined sets of values that reflect years worth of process and analytic investment.

As a result, users of Ed-Fi APIs should expect to see enumeration sets with mixed namespaces.

### Use Case-Specific Information

Ed-Fi API specifications and certifications capture current community expectations as to what enumeration sets to use. Specifications in use can be found here:

> [Ed-Fi Data Exchange Standards Home](https://edfi.atlassian.net/wiki/spaces/EFDS/overview)

Reviewer: I'm not sure what to do about this. These documents are ancient and should not be used - they refer to versions of the Data Standard that are no longer supported.

The RFC Home below are similarly not very useful as far as I'm concerned, because so out of date.

For now I'll leave these Confluence links in place, though I really don't want a technology provider to read them.

Specifications in a "Draft" Request for Comment (RFC) state (which are also commonly found) can be found here:

[Ed-Fi RFC Home](https://edfi.atlassian.net/wiki/spaces/EFDSRFC/overview)

Certifications for these specifications and others are listed here:

> [Available Certifications](https://edfi.atlassian.net/wiki/spaces/EDFICERT/pages/23692586/Available+Certifications)

Reviewer: ```suggestion
[Available Certifications](/partners/certification)
```

Author: updated.

### Example: Assessment Outcomes API

As an example, let's look at the Assessment Outcomes API.

> In this API (which is defined in [ED-FI RFC 15 - ASSESSMENT OUTCOMES API](https://edfi.atlassian.net/wiki/spaces/EFDSRFC/pages/25362990/ED-FI+RFC+15+-+ASSESSMENT+OUTCOMES+API)), there are no requirements as to what enumeration sets are required. However, in the certification for this API, only certain enumerations are permitted to be tech-provider-supplied and others must be Ed-Fi-governed values.  There are currently three major versions of the Data Standard available so look at these locations for examples:

Reviewer: This is an example of a very dangerous link. Someone who follows this and tries to build an Assessment Outcomes API will be building on Data Standard 2.2, which was deprecated years ago.

We should change to this link, which is at least on Data Standard 3 and not so far off as the Data Standard 2 doc.

```suggestion
In this API (which is defined in [ED-FI RFC 15 - ASSESSMENT OUTCOMES API](https://edfi.atlassian.net/wiki/spaces/EFDS/pages/17727736/Ed-Fi+Assessment+Outcomes+API+for+Suite+3)), there are no requirements as to what enumeration sets are required. However, in the certification for this API, only certain enumerations are permitted to be tech-provider-supplied and others must be Ed-Fi-governed values.  There are currently three major versions of the Data Standard available so look at these locations for examples:
```

Author: updated.

> * [Ed-Fi Assessment Outcomes API for Suite 3 Certification - Ed-Fi Certification - Ed-Fi TechDocs (atlassian.net)](https://edfi.atlassian.net/wiki/spaces/EDFICERT/pages/23697946/Ed-Fi+Assessment+Outcomes+API+for+Suite+3+Certification)
> * [Ed-Fi Assessment Outcomes API for Data Standard v4 Certification - Ed-Fi Certification - Ed-Fi TechDocs (atlassian.net)](https://edfi.atlassian.net/wiki/spaces/EDFICERT/pages/23701219/Ed-Fi+Assessment+Outcomes+API+for+Data+Standard+v4+Certification)
> * [Ed-Fi Assessment Outcomes API for Data Standard v5 Certification - Ed-Fi Certification - Ed-Fi TechDocs (atlassian.net)](https://edfi.atlassian.net/wiki/spaces/EDFICERT/pages/23702691/Ed-Fi+Assessment+Outcomes+API+for+Data+Standard+v5+Certification)

Reviewer: ```suggestion
* [Ed-Fi Assessment Outcomes API for Suite 3 Certification, **v4**](/partners/certification/available-certifications/assessment-v4/)
* [Ed-Fi Assessment Outcomes API for Data Standard, **v5**](/partners/certification/available-certifications/assessment-v5/)
```
